singledispatch/__init__.py

```diff
@@ -232,8 +247,13 @@ def register(cls, func=None):
         return func
 
     def wrapper(*args, **kw):
+        if not args:
```

Comment:
I'd like to see these unrelated changes in a separate commit from the PR. I'll cherry-pick the change if it's separate. Otherwise, I'll commit it separately in main.

Reply:
cherry-picked to main in 01953ff.